Upstream: submitted; https://lists.gnu.org/archive/html/screen-devel/2014-07/msg00002.html

From d340b02cbedb7e6e726559f26dc09d3b8b104ec5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20S=C5=82awi=C5=84ski?= <amade@asmblr.net>
Date: Tue, 8 Jul 2014 14:55:25 +0200
Subject: [PATCH] Do not link against libelf.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I'd like to upstream the patch to drop linking dependency to libelf. It is
probably some leftover from old versions of gnu screen and some
distributions downstream upply it for years.

http://pkgs.fedoraproject.org/cgit/screen.git/tree/screen-4.0.3-libs.patch
(Fedora)
http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/app-misc/screen/files/4.0.2-no-libelf.patch?revision=1.1&view=markup
(Gentoo)
https://galileo.mailstation.de/gerrit/#/c/1275/ (Exherbo)

From: Ivan Diorditsa <ivan.diorditsa@gmail.com>
Signed-off-by: Amadeusz Sławiński <amade@asmblr.net>
---
 screen-4.3.0/configure.ac | 8 --------
 1 file changed, 8 deletions(-)

diff --git screen-4.3.0/configure.ac screen-4.3.0/configure.ac
index 87fd43b..e5d8676 100644
--- screen-4.3.0/configure.ac
+++ screen-4.3.0/configure.ac
@@ -201,14 +201,6 @@ AC_EGREP_CPP(yes,
 #endif
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
-oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
-AC_CHECKING(SVR4)
-AC_TRY_LINK([#include <utmpx.h>
-],,
-[AC_CHECK_HEADER(dwarf.h, AC_DEFINE(SVR4) AC_DEFINE(BUGGYGETLOGIN),
-[AC_CHECK_HEADER(elf.h, AC_DEFINE(SVR4) AC_DEFINE(BUGGYGETLOGIN))])]
-,LIBS="$oldlibs")
 AC_CHECK_HEADERS([stropts.h string.h strings.h])
 
 AC_CHECKING(for Solaris 2.x)
-- 
2.0.0

